crates/runtime/src/backtrace/aarch64.rs

```diff
@@ -0,0 +1,19 @@
+pub unsafe fn get_next_older_pc_from_fp(fp: usize) -> usize {
+    *(fp as *mut usize).offset(1)
+}
+
+pub unsafe fn get_next_older_fp_from_fp(fp: usize) -> usize {
+    *(fp as *mut usize)
+}
+
+pub fn reached_entry_sp(fp: usize, first_wasm_sp: usize) -> bool {
+    fp == first_wasm_sp - 16
+}
+
+pub fn assert_entry_sp_is_aligned(sp: usize) {
+    assert_eq!(sp % 16, 0, "stack should always be aligned to 16");
+}
+
+pub fn assert_fp_is_aligned(fp: usize) {
+    assert_eq!(fp % 16, 0, "stack should always be aligned to 16");
```

Comment:
Technically there is no alignment requirement for the frame pointer - in fact, if my reading of the AAPCS64 is correct (`The location of the frame record within a stack frame is not specified.`), then it could even be completely unaligned (or at the very least aligned on 8 bytes)! However, it is probably still useful to leave this check as is.